tests/psi4numpy/rhf_hessian/input.py

```diff
@@ -0,0 +1,284 @@
+"""
+This script calculates nuclear Hessians of RHF Wavefunction using
+derivatives of one and two electron integrals obtained from PSI4
+
+Reference: "Derivative studies in hartree-fock and møller-plesset theories",
+J. A. Pople, R. Krishnan, H. B. Schlegel and J. S. Binkley
+DOI: 10.1002/qua.560160825
+"""
+
+__authors__ = "Ashutosh Kumar"
+__credits__ = ["Ashutosh Kumar"]
+
+__copyright__ = "(c) 2014-2017, The Psi4NumPy Developers"
+__license__ = "BSD-3-Clause"
+__date__ = "2017-12-17"
+
+import time
+import numpy as np
+np.set_printoptions(precision=15, linewidth=200, suppress=True)
+import psi4
+
+psi4.set_memory(int(1e9), False)
+psi4.core.set_output_file('output.dat', False)
+psi4.core.set_num_threads(4)
+
+mol = psi4.geometry("""
+O
+H 1 1.1
+H 1 1.1 2 104
+symmetry c1
+""")
+
+psi4.core.set_active_molecule(mol)
+
+options = {'BASIS':'STO-3G', 'SCF_TYPE':'PK',
+           'E_CONVERGENCE':1e-10,
+           'D_CONVERGENCE':1e-10
+           }
+
+psi4.set_options(options)
+
+rhf_e, wfn = psi4.energy('SCF', return_wfn=True)
+
+# Assuming C1 symmetry    
+occ = wfn.doccpi()[0]
+nmo = wfn.nmo()
+vir = nmo - occ
+
+C = wfn.Ca_subset("AO", "ALL")
+npC = np.asarray(C)
+
+mints = psi4.core.MintsHelper(wfn.basisset())
+H_ao = np.asarray(mints.ao_kinetic()) + np.asarray(mints.ao_potential())
+
+# Update H, transform to MO basis 
+H = np.einsum('uj,vi,uv', npC, npC, H_ao)
+
+# Integral generation from Psi4's MintsHelper
+MO = np.asarray(mints.mo_eri(C, C, C, C))
+# Physicist notation    
+MO = MO.swapaxes(1,2)
+
+F = H + 2.0 * np.einsum('pmqm->pq', MO[:, :occ, :, :occ])
+F -= np.einsum('pmmq->pq', MO[:, :occ, :occ, :])
+natoms = mol.natom()
+cart = {}; cart[0] = '_X'; cart[1] = '_Y'; cart[2] = '_Z'
```

Comment:
This is only used as an array, not a dict, as far as I can tell.  Maybe `cart = ['_X', '_Y', '_Z']` is a little more Pythonic in that case.